### ISSUE TITLE
Fichier pour les membres

### ISSUE THREAD

**Issue by fxleblanc** [ClubCedille/canoe-beton#82]
## Format
- nom
- role
- chemin vers la photo(pourrait même être déterminé à partir du nom)
## git

Est-ce qu'on garderait le fichier dans l'historique git.

**Comment by mikefaille** [comment]
«Est-ce qu'on garderait le fichier dans l'historique git.»
Je crois que ça serait l'idéal; c'est un peu comme ça que fonctionne Jekyll.

**Comment by fxleblanc** [comment]
Je suis d'accord. Pour le type de fichier, je proposerais du json. C'est moins déclaratif mais plus simple à lire que du markdown d'un point de vue javascript. Au final, quand tu veux modifier le fichier, tu vas juste copier coller sans trop te soucier de la beauté du fichier.

**Comment by mikefaille** [comment]
Ah, je comprends. 

Qu'est-ce que tu penses d'une librairie Javascript Markdown à HTML ? 
- https://github.com/evilstreak/markdown-js .
- https://github.com/chjj/marked
- http://strapdownjs.com/
- https://simonwaldherr.github.io/micromarkdown.js/

**Comment by fxleblanc** [comment]
C'est possible et je suis sûr qu'il en existe des bonnes. Mon point est que ça fait une librairie de plus pour pas grand chose. Le json est supporté nativement et dans l'état des choses, uniquement des programmeurs avec une compréhension du json vont modifier le fichier. Le markdown serait utile pour une audience sans compréhension du json(et encore, le copy paste est pas mal simple, même avec du json).

**Comment by mikefaille** [comment]
C'est vrai, la plupart des programmeurs savent comment modifier les JSON. Je crois aussi que même pour les programmeurs, l'édition de texte est plus naturelle avec du markdown. Même qu'à mon avis, n'importe qui en génie peut modifier du markdown directement avec le UI de github avec les outils de formatage des champs de textes sur Github.

**Comment by fxleblanc** [comment]
Je suis d'accord que le markdown est plus "intéressant" que le JSON mais l'édition par Github n'est pas vraiment possible puisqu'il faut ajouter/supprimer des images et qu'au final, le programmeur va devoir de toute façon aller en local. Par contre, pour un site github.io sans contenu autre que du texte, c'est une bonne manière de faire. Au final, je resterais avec le JSON pour la simplicité.

**Comment by mikefaille** [comment]
btw, peu importe la décision que tu vas prendre, le résultat sera super intéressant. D'ailleurs, s'il y a une itération à faire même dans le cas de ma suggestion, tu auras tout de même à manipuler un résultat en json. Note que pour même les vielles versions de wordpress il fallait mettre le chemin de l'image. Puis, concernant l'upload de l'image; d'ailleurs, c'est possible faire du drag-and-drop tout comme wordpress ie : https://github.com/blog/2105-upload-files-to-your-repositories.

**Comment by fxleblanc** [comment]
J'ai bien hâte d'avoir la feature de github pour upload les fichiers. À ce moment-là, il serait intéressant d'utiliser du markdown puisqu'il sera possible à quelqu'un sans expérience d'aller jouer dans le markdown et d'upload des fichiers. Cependant, à moins d'un script automatisé, le site ne sera pas redéployé et cela demendera tout de même quelqu'un responsable du déploiement.

**Comment by mikefaille** [comment]
Content que tu t'intéresses à ça. C'est possible d'avoir un interprétation live du markdown ce qui éviterait le redéployement. Cette lib est un bel example : https://github.com/evilstreak/markdown-js
